dts/bindings/ethernet/nxp,enet-mac.yaml

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP ENET MAC/L2 Device
+
+compatible: "nxp,enet-mac"
+
+include: ["ethernet-controller.yaml", "ethernet,fixed-link.yaml", "pinctrl-device.yaml"]
+
+properties:
+  interrupts:
+    required: true
+
+  nxp,interface-mode:
+    required: true
+    type: string
+    enum:
+      - "mii"
+      - "rmii"
+    description: |
+      The interface that the PHY is connected to the ENET
```

Comment:
Can we avoid this property and use what is provide in the phy node. Seems like the user will have to specify the same value in 2 places.

Reply:
no, it needs to be in both places, the phy and ethernet IP are independent devices

Reply:
So they could be different?

Reply:
it wouldn't make sense for them to be different but both devices need to be configured for the correct interface. The fact that both the ethernet controller and the phy in this case have the functionality to configure the interface is just a coincidence. this ethernet controller could be used with a phy that is not configurable and needs to choose the right one, and the phy could be used with an ethernet controller that is not configurable and needs to be set up for the right interface. both of these bindings need this property. in the case of these EVKs, they are modular parts of the same system, but that is not the same as being the same device

Reply:
to align with linux it has to be a property of the ethernet controller (MAC), to align with what zephyr needs as far as initialization flow, it has to also be on this specific phy binding, because the phy cannot get information from the MAC node in this case without some complicated runtime callbacks, because there would be a circular reference in DT which is not allowed